src/renderers/WebGLRenderer.js

```diff
@@ -2176,7 +2176,23 @@ function WebGLRenderer( parameters = {} ) {
 
 	this.initTexture = function ( texture ) {
 
-		textures.setTexture2D( texture, 0 );
+		if ( texture.isCubeTexture ) {
+
+			textures.setTextureCube( texture, 0 );
+
+		} else {
+
+			if ( texture.isData3DTexture ) {
```

Comment:
Would it make sense to also support `DataArrayTexture`?

Reply:
:thinking: yeah.
At this point, It makes sense to already include all the texture types.

Reply:
Please also use `else if`s instead of nested ifs.